asapdiscovery-data/asapdiscovery/data/state_expanders/state_expander.py

```diff
@@ -0,0 +1,51 @@
+import abc
+from enum import Enum
+
+from asapdiscovery.data.schema_v2.ligand import Ligand
+from pydantic import BaseModel, Field
+
+
+class StateExpanderType(str, Enum):
+    STEREO = "STEREO"
+    TAUTOMER = "TAUTOMER"
+    PROTOMER = "PROTOMER"
+
+
+class StateExpanderBase(BaseModel):
+    input_ligands: list[Ligand] = Field(..., description="The ligand to be expanded")
+    expander_type: StateExpanderType = Field(
+        ..., description="The type of expander to use"
+    )
+
+    @abc.abstractmethod
+    def _expand(self) -> list["StateExpansion"]:
+        ...
+
+    def expand(self) -> list["StateExpansion"]:
+        return self._expand()
+
+
+class StateExpansion(BaseModel):
+    parent: Ligand = Field(..., description="The parent ligand")
+    children: list[Ligand] = Field(
+        ..., description="The children ligands resulting from expansion"
+    )
+    expander: StateExpanderBase = Field(
```

Comment:
Have this be the actual expander used like `StereoExpander` to capture the runtime settings or a dict of that class as long as it has all of the settings.